tests/driver.C

```diff
@@ -134,15 +140,17 @@ int main(int argc, char ** argv)
   // Add all tests which match the re to the runner object.
   libMesh::out << "Will run the following tests:" << std::endl;
   const int n_tests_added =
-    add_matching_tests_to_runner(registry.makeTest(),
+    add_matching_tests_to_runner(suite,
                                  allow_regex_string, allow_regex,
                                  deny_regex_string, deny_regex,
                                  runner, rejects);
   if (n_tests_added >= 0)
     libMesh::out << "--- Running " << n_tests_added << " tests in total." << std::endl;
+  if (n_tests_added != -12345)
+    owned_suite.reset(suite);
```

Comment:
I guess I don't understand why we don't need to clean up `suite` when there's no tests added? From a surface level reading of the code it just looks like `registry.makeTest()` returns a dumb pointer whose lifetime we are expected to manage, and we were always leaking it before...

Reply:
This should clean up `suite` when there's no tests added.  That'll return `n_tests_added == 0`, and `0 != -12345`, so we put `suite` in our `unique_ptr` and it gets cleaned up.